config_test.go

```diff
@@ -144,3 +145,72 @@ func TestConfigWithMissingAttributes(t *testing.T) {
 		})
 	}
 }
+
+func makeSamplerCountingHook() (func(zapcore.Entry, zapcore.SamplingDecision),
+	*atomic.Int64, *atomic.Int64) {
```

Comment:
can you name the returns? we also can do, `dropped, sampled *atomic.Int64` to avoid duplicating the type.

And a super nit: The way this declaration is broken up over multiple lines looks a little odd. We don't tend to force line length < 80, can we put the whole declaration on one line?